File: code/jvm/im-repository/src/main/kotlin/im/repository/pagination/PaginationRequest.kt
package im.repository.pagination

/**
 * Represents a pagination request.
 *
 * @param offset the offset of the first element to return
 * @param limit the maximum number of elements to return
 * @param getCount whether to count the total number of elements
 */
data class PaginationRequest(
    val offset: Long,
    val limit: Int,
    val getCount: Boolean = true,
)

File: code/client/web/src/Domain/pagination/PaginationRequest.ts
/**
 * Represents a pagination request.
 * @param page The page number.
 * @param size The number of items per page.
 * @param getCount Indicates whether to return the total count of items.
 */
export interface PaginationRequest {
    offset: number | null;
    limit: number | null;
    getCount: boolean | null;
}
